windows/setup.ps1

```diff
@@ -77,29 +81,80 @@ foreach ($adapter in $ActiveNetcfg)
     $formattednetcfg += $thisadapter;$thisadapter = $null
 }
 
+$cpu_list = @( )
+for ($i=1; $i -le ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors); $i++) {
+    $cpu_list += $win32_cpu.Manufacturer
+    $cpu_list += $win32_cpu.Name
+}
+
 Set-Attr $result.ansible_facts "ansible_interfaces" $formattednetcfg
 
 Set-Attr $result.ansible_facts "ansible_architecture" $win32_os.OSArchitecture
 
-Set-Attr $result.ansible_facts "ansible_hostname" $env:COMPUTERNAME;
-Set-Attr $result.ansible_facts "ansible_fqdn" "$([System.Net.Dns]::GetHostByName((hostname)).HostName)"
+Set-Attr $result.ansible_facts "ansible_bios_date" $win32_bios.ReleaseDate.ToString("MM/dd/yyyy")
+Set-Attr $result.ansible_facts "ansible_bios_version" $win32_bios.SMBIOSBIOSVersion
+Set-Attr $result.ansible_facts "ansible_hostname" $env:COMPUTERNAME
+Set-Attr $result.ansible_facts "ansible_fqdn" ($ip_props.Hostname + "." + $ip_props.DomainName)
+Set-Attr $result.ansible_facts "ansible_processor" $cpu_list
+Set-Attr $result.ansible_facts "ansible_processor_cores" $win32_cpu.NumberOfCores
+Set-Attr $result.ansible_facts "ansible_processor_count" $win32_cs.NumberOfProcessors
+Set-Attr $result.ansible_facts "ansible_processor_threads_per_core" ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors / $win32_cpu.NumberOfCores)
+Set-Attr $result.ansible_facts "ansible_processor_vcpus" ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors)
+Set-Attr $result.ansible_facts "ansible_product_name" $win32_cs.Model.Trim()
+Set-Attr $result.ansible_facts "ansible_product_serial" $win32_bios.SerialNumber
+Set-Attr $result.ansible_facts "ansible_product_version" ([string] $win32_cs.SystemFamily)
```

Comment:
I expected that to come from the BIOS info (actually DMI), but it wasn't available there. Will look this evening for other places with the same content.